mmseg/core/evaluation/eval_hooks.py

```diff
@@ -13,31 +13,56 @@ class EvalHook(_EvalHook):
         by_epoch (bool): Determine perform evaluation by epoch or by iteration.
             If set to True, it will perform by epoch. Otherwise, by iteration.
             Default: False.
-        efficient_test (bool): Whether save the results as local numpy files to
-            save CPU memory during evaluation. Default: False.
     Returns:
         list: The prediction results.
     """
 
     greater_keys = ['mIoU', 'mAcc', 'aAcc']
 
-    def __init__(self, *args, by_epoch=False, efficient_test=False, **kwargs):
```

Comment:
Keep the API the same, add deprecate warning.

Reply:
Add deprecated warning and modify it to compact old version.